lib/engine/part/node.rb

```diff
@@ -57,6 +57,7 @@ def walk(
           node_path.walk(
             visited: visited_paths,
             skip_paths: skip_paths,
+            skip_track: skip_track,
```

Comment:
Was this needed for something specific to 1858? If it's a bug fix for general graph walking, let's get this in a separate PR.

Reply:
It's a bug fix, though it might not be possible to trigger this bug in other games. In 1858 you're allowed to build broad gauge and narrow gauge track so that they meet at a hex edge, but it's not possible to trace a route or run any trains down a link like this. So you can have something like node, broad gauge track, narrow gauge track, node. As the skip_track setting was not being passed down it was possible for this route to be walked without objecting to the gauge change.

I'll create a pull request that is just for this change.

Reply:
I've created pull request #8612 for this fix.

Reply:
This pull request has been merged.